src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -424,7 +390,8 @@ FancyZones::OnKeyDown(PKBDLLHOOKSTRUCT info) noexcept
     //    CycleActiveZoneSet(info->vkCode);
     //    return false;
     //}
-    if (m_dragEnabled && shift)
+
+    if (m_windowMoveHandler.IsDragEnabled() && shift)

```

Comment:
is it okay that we don't mutex-protect some of the `m_windowMoveHandler` method calls?

Reply:
Correct me if I'm wrong: the WindowMoveHandler should only be called by the main thread, if it's called by any other thread that is an implementation bug. Maybe we should add asserts.

Reply:
I didn't add mutexes in WindowMoveHandler since it also uses data from FancyZones (`m_zoneWindowMap`) so it should be protected by caller. 
I think we should better keep all data synchronization in one place (`FancyZones`).

And of course mutex has to be there, I'll add it